src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -400,7 +410,100 @@ private IObjectCreationExpression CreateBoundObjectCreationExpressionOperation(B
+            bool isInvalid = boundCollectionInitializerExpression.HasErrors;
+            SyntaxNode syntax = boundCollectionInitializerExpression.Syntax;
+            ITypeSymbol type = boundCollectionInitializerExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundCollectionInitializerExpression.ConstantValue);
+            return new LazyObjectOrCollectionInitializerExpression(initializers, isInvalid, syntax, type, constantValue);
+        }
+
+        private IMemberReferenceExpression CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember)
+        {
+            Lazy<IOperation> instance = new Lazy<IOperation>(() => new InstanceReferenceExpression(InstanceReferenceKind.Implicit, boundObjectInitializerMember.HasErrors, syntax: boundObjectInitializerMember.Syntax, type: boundObjectInitializerMember.MemberSymbol.ContainingType, constantValue: default(Optional<object>)));
+            bool isInvalid = boundObjectInitializerMember.HasErrors;
+            SyntaxNode syntax = boundObjectInitializerMember.Syntax;
+            ITypeSymbol type = boundObjectInitializerMember.Type;
+            Optional<object> constantValue = ConvertToOptional(boundObjectInitializerMember.ConstantValue);
+
+            switch (boundObjectInitializerMember.MemberSymbol.Kind)
+            {
+                case SymbolKind.Field:
+                    var field = (FieldSymbol)boundObjectInitializerMember.MemberSymbol;
+                    return new LazyFieldReferenceExpression(field, instance, field, isInvalid, syntax, type, constantValue);
+                case SymbolKind.Event:
+                    var eventSymbol = (EventSymbol)boundObjectInitializerMember.MemberSymbol;
+                    return new LazyEventReferenceExpression(eventSymbol, instance, eventSymbol, isInvalid, syntax, type, constantValue);
+                case SymbolKind.Property:
+                    var property = (PropertySymbol)boundObjectInitializerMember.MemberSymbol;
+                    Lazy<ImmutableArray<IArgument>> argumentsInEvaluationOrder;
+                    if (!boundObjectInitializerMember.Arguments.Any())
+                    {
+                        // Simple property reference.
+                        argumentsInEvaluationOrder = new Lazy<ImmutableArray<IArgument>>(() => ImmutableArray<IArgument>.Empty);
+                    }
+                    else
+                    {
+                        // Indexed property reference.
+                        argumentsInEvaluationOrder = new Lazy<ImmutableArray<IArgument>>(() =>
+                        {
+                            return DeriveArguments(boundObjectInitializerMember.HasErrors,
+                                boundObjectInitializerMember.BinderOpt,
+                                property,
+                                property.SetMethod,
```

Comment:
`CreateBoundIndexerAccessOperation()` uses `GetOwnOrInheritedSetMethod()`. Are there cases where that will give a different accessor here? If so, consider extracting a helper method that can be used by the factory methods.